lib/active_model/serializer/adapter/json.rb

```diff
@@ -47,6 +31,32 @@ def serializable_hash(options = {})
       def fragment_cache(cached_hash, non_cached_hash)
         Json::FragmentCache.new().fragment_cache(cached_hash, non_cached_hash)
       end
+
+      private
+
+      def populate_hash_for_array_serializer(name, association, opts)
+        return unless association.respond_to?(:each)
```

Comment:
IMHO I'm not sure about moving the conditionals to inside this private methods.
It just not seems to me that this conditional handling would be their responsibility. Running the same conditional twice is a smell of that.